pkg/chain/chain.go

```diff
@@ -0,0 +1,7 @@
+package chain
+
+// IChain is an abstraction for a cryptocurrency
+// It only enables creating raw transactions requests
+type IChain interface {
+	NewRequest(txHex string) ([]byte, error)
```

Comment:
I like this interface. The only thing that I see that will be missing is maybe adding the RPC url endpoint. Binance (and any cosmos fork) has this URL path for `POST`ing a transaction:

`http://<node ip>:26656/broadcast_tx_async?tx=`

So we can either add the full URL path from above as the value of `rpcURL` of `currency.toml` or try to format the URL to something we want like [here](https://github.com/hashcloak/Meson-plugin/blob/binance-capabilities/pkg/proxy/proxy.go#L112). I prefer to keep it simple and have less code so this is good for me. We can add more fancy code in the future.

Reply:
So, anything like this that is specific to a chain should just be added in a `CosmosChain` struct that implements the `IChain` interface. This way it abstracts away how the inner workings of sending a new request is to a Cosmos chain.